HW6 MVC шаблон, работа с базой данных и страница добавления в индекс

r-i-besedin opened the pull request:
Добавил возможность работы с бд, страницу для добавления в индекс и mvc-шаблон

r-i-besedin committed:
MVC шаблон, работа с базой данных и страница добавления в индекс

Добавил возможность работы с бд, страницу для добавления в индекс и интерфейсы к модели

r-i-besedin commented:
Это не конечный, надо еще доделать несколько вещей, добавить обработку слов, добавление как-то сделать, чтобы можно было через веб интерфейс добавлять одновременно, но вариант уже рабочий

r-i-besedin committed:
Подправил шаблоны и перенес шаблоны и статик файлы в папку view

tariel-x commented:
Всё ок, только не делайте panic внутри кода. Лучше верните ошибку в main и там с ней разберитесь. Вообще, на самом деле, мы тут все используем панику, но на деле она не приветствуется. Обычно выходят из программы другими способами, с логами, с завершением открытых http-соединений и т.д.

tariel-x commented:
А ещё надо как-то опционально делать очистку-загрузку БД. На деле ведь никто после перезапуска не будет перезагружать бд, она для того и создана, чтоб персистентно хранить данные.

Можете посмотреть в сторону миграций БД.

tariel-x commented:
И 3-й момент: вставка записей по одной происходит очень медленно. Эффективней будет вставка пачкой: https://godoc.org/github.com/go-pg/pg#example-DB-Insert-BulkInsert . Или же можно открыть транзацию, сделать n вставок, потом сделать Commit транзакции. Если вставлять по отдельности, то каждый раз будет открываться новая. Учитывая, что они изолированные, это неэффективно.

r-i-besedin committed:
Стеммер и улучшение разбиения по словам

Улучшил функцию разбиения по словам и добавил стемминг. Теперь можно парсить даже html-странички со сложными тегами 🙈

r-i-besedin committed:
Улучшена работа с БД, стемминг и паники

Теперь запросы на добавление отправляются пачками, существенно снизилось время инициализации, паники вроде везде убрал, оставил только действительно в самых важных местах, стемминг немного неправильно использовал, исправил, теперь всё работает стабильно

tariel-x commented:
:+1:
